client/Android/Studio/freeRDPCore/src/main/java/com/freerdp/freerdpcore/presentation/ApplicationSettingsActivity.java

```diff
@@ -81,13 +81,15 @@ public void onCreate(Bundle savedInstanceState) {
 
         @Override
         public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, String key) {
-            final String clientNameKey = getString(R.string.preference_key_client_name);
-
-            get(getActivity());
-            if (key.equals(clientNameKey)) {
-                final String clientNameValue = sharedPreferences.getString(clientNameKey, "");
-                EditTextPreference pref = (EditTextPreference) findPreference(clientNameKey);
-                pref.setText(clientNameValue);
+            if (isAdded()) {
```

Comment:
Why only change when the key is added? May be changed/deleted/...

Reply:
It prevents crashing the app due the exception: "Fragment ClientPreferenceFragment not attached to Activity" Please see solution: https://stackoverflow.com/questions/10919240/fragment-myfragment-not-attached-to-activity